File: Drawer/Drawer/ViewController.m
//
//  ViewController.m
//  Drawer
//
//  Created by Tengfei on 16/11/16.
//  Copyright © 2016年 tengfei. All rights reserved.
//

#import "ViewController.h"

#import "DrawerController.h"

#import "LeftDemoViewController.h"
#import "CenterDemoViewController.h"
#import "RightDemoViewController.h"

@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view, typically from a nib.
    
    //当一个控制器的View添加到另一个控制器的View上的时候,那此时View所在的控制器也应该成为上一个控制器的子控制器.
    CenterDemoViewController *vc1 = [[CenterDemoViewController alloc] init];
    vc1.view.frame = self.mainView.bounds;
    [self.mainView addSubview:vc1.view];
    [self addChildViewController:vc1];
    
    
    LeftDemoViewController *vc2 = [[LeftDemoViewController alloc] init];
    vc2.view.backgroundColor = [UIColor redColor];
    vc2.view.frame = self.mainView.bounds;
    [self.leftView addSubview:vc2.view];
    [self addChildViewController:vc2];
    
    RightDemoViewController *vc3 = [[RightDemoViewController alloc] init];
    vc3.view.backgroundColor = [UIColor blackColor];
    vc3.view.frame = self.mainView.bounds;
    [self.rightView addSubview:vc3.view];
    [self addChildViewController:vc3];
    
}


- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}


@end

File: FileMaster/FileMaster/ViewController.swift
//
//  ViewController.swift
//  FileMaster
//
//  Created by Fengtf on 16/2/24.
//  Copyright © 2016年 ftf. All rights reserved.
//

import UIKit


 
class ViewController: UIViewController {

    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }

    
    
    
    func readeFile(sender: UIButton) {
        let path = ( NSHomeDirectory() as NSString ).stringByAppendingPathComponent("Documents")
        let fm = NSFileManager.defaultManager()
        
        var array:[String] = []
        do {
            array = try fm.contentsOfDirectoryAtPath(path)
        }catch {
            print(error)
        }
        for name in array {
            print("name:\(name),,,path:\(path.stringByAppendingString("/\(name)"))")
        }
    }
    
    func writeFile(sender: UIButton) {
        let path = ( NSHomeDirectory() as NSString ).stringByAppendingPathComponent("Documents")
        let fm = NSFileManager.defaultManager()
        let fileName = path.stringByAppendingString("/rrr.png")
        
        let image = createImage()
        let data = UIImagePNGRepresentation(image)
        fm.createFileAtPath(fileName, contents: data, attributes: nil)
    }
    
    func createImage() -> UIImage{
        UIGraphicsBeginImageContextWithOptions(CGSize(width: 300, height: 320), false, 0.0)
        let ctx = UIGraphicsGetCurrentContext()
        let imageRect = CGRect(x: 0.0, y: 0.0, width: 300, height: 320)
        //        UIColor.brownColor().setFill()
        CGContextFillRect(ctx, imageRect)
        let image = UIGraphicsGetImageFromCurrentImageContext()
        UIGraphicsEndImageContext()
        return image
    }


}

